service/s3/unmarshal.go

```diff
@@ -0,0 +1,109 @@
+package s3
+
+import (
+	"bytes"
+	"encoding/xml"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	restlegacy "github.com/aws/aws-sdk-go-v2/private/protocol/rest"
+)
+
+/*
+   GetObject operation doesn't support 200 status code with errors.
+
+*/
+
+// protoGetObjectUnmarshaler defines unmarshaler for ProtoGetObject operation
+type protoGetObjectUnmarshaler struct {
+	output *GetObjectOutput
+}
+
+// unmarshalOperation is the top level method used with a handler stack to unmarshal an operation response
+// This method calls appropriate unmarshal shape functions as per the output shape and protocol used by the
+// service.
+func (u protoGetObjectUnmarshaler) unmarshalOperation(r *aws.Request) {
+	// isRequestError checks if operation response returned a error response
+	if isRequestError(r) {
+		// if startToken is nil, it would mean there is no xml response body
+		if r.HTTPResponse.Body == nil {
+			// fall back to status code error message
```

Comment:
Probably want to ensure that `r.HTTPResponse.Body.Close` is called before this method returns.